### PR TITLE
Joke Component

### PR DESCRIPTION
**Pull request by micsoni**
Added Joke component and tested its behavior with unit tests. bumped @testing-library and added "jest-environment-jsdom-sixteen" to the dependencies, so waiFor would work.
I also added a test to check if the App component renders.

### CODE REVIEW

File: src/components/Joke.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import Joke from "./Joke";
+
+describe("Joke Component", () => {
+  it("should load only setup on intial render", () => {
+    const { getByText } = render(
+      <Joke setup="setup test" punchline="punchline test" />
+    );
+    const jokeElement = getByText("setup test");
+    expect(jokeElement).toBeTruthy();
```
Comment (vloth) [inline src/components/Joke.test.js]: `toBeInTheDocument` might be a clearer assertion here

File: src/components/Joke.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  });
+
+  it("should load punchline after 100 milliseconds", async () => {
+    const { getByText } = render(
+      <Joke setup="setup test" punchline="punchline test" delay={100} />
+    );
+
+    await waitFor(() => {
+      expect(getByText("punchline test")).toBeTruthy();
```
Comment (vloth) [inline src/components/Joke.test.js]: Same here, see above

**Review by vloth — CHANGES_REQUESTED** [review]

File: src/components/Joke.test.js (hunk 1 above)
Comment (micsoni) [reply src/components/Joke.test.js]: I agree. Done!

File: src/components/Joke.test.js (hunk 2 above)
Comment (micsoni) [reply src/components/Joke.test.js]: Also done.

**Review by vloth — APPROVED** [review]
LGTM

**Pull request merged**